google-cloud-pubsub/lib/google/cloud/pubsub/subscriber.rb

```diff
@@ -167,13 +168,7 @@ def stop
         # @return [Subscriber] returns self so calls can be chained.
         #
         def wait! timeout = nil
-          wait_pool = synchronize do
-            @stream_pool.map do |stream|
-              Thread.new { stream.wait! timeout }
-            end
-          end
-          wait_pool.map(&:join)
-
+          @final_stopped_thread&.join timeout
```

Comment:
One more thing I realize, this is actually a behavior regression. Previously, if you called `wait!` without (or before) calling `stop`, the `wait!` call would still block until someone called `stop` and everything finished. Now, we return immediately if no one has called `stop` yet. I think we might need to preserve the previous behavior.

Reply:
Got it, sounds good.

Reply:
@dazuma Updated to preserve the previous behavior of `wait!`.